crowbar_framework/app/controllers/api/upgrade_controller.rb

```diff
@@ -64,6 +64,24 @@ def services
     end
   end
 
+  api :POST, "/api/upgrade/controllers", "Orchestrate the upgrade of controller nodes"
```

Comment:
but then the user can't really choose which controller to upgrade. or do you want to provide some parameter which controller to upgrade? if not, then that would only make sense to implement internally (so not exposed through the api). would be also a question for the UI/UX team?